CMake users using jaeger exporter are forced to know about internal opentelemetry details

**Describe your environment** 
Linux Fedora 36, Opentelemetry built with shared libs, building with CMake

**Steps to reproduce**
Build code that uses the jaeger exporter using a CMake file like:
```
target_link_libraries(qpid-proton-cpp LINK_PRIVATE ...
opentelemetry-cpp::trace opentelemetry-cpp::jaeger_trace_exporter opentelemetry-cpp::ostream_span_exporter)
```
**What is the expected behavior?**
Build with no failure!

**What is the actual behavior?**
```
CMake Warning (dev) at cpp/CMakeLists.txt:122 (add_library):
  Policy CMP0028 is not set: Double colon in target name means ALIAS or
  IMPORTED target.  Run "cmake --help-policy CMP0028" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target "qpid-proton-cpp" links to target "CURL::libcurl" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
This warning is for project developers.  Use -Wno-dev to suppress it.
```
...
```
/usr/bin/ld: cannot find -lCURL::libcurl
collect2: error: ld returned 1 exit status
```

**Additional context**
There is a work around for this issue which is to use ```find_package(CURL REQUIRED)``` in the CMake files. However this makes no real sense as using libcurl is an internal detail of the opentelemetry-cpp sdk jaager exporter via the http_curl library and has no reason to be exposed to the user.

I'm also led to think it is an oversight due to it not being documented.

The simple fix that works for me is to make the CURL::libcurl link PRIVATE in ```ext/src/http/client/curl/CMakeLists.txt```

--- a/ext/src/http/client/curl/CMakeLists.txt
+++ b/ext/src/http/client/curl/CMakeLists.txt
@@ -8,13 +8,17 @@ if(CURL_FOUND)
                         PROPERTIES EXPORT_NAME http_client_curl)
 
   if(TARGET CURL::libcurl)
-    target_link_libraries(opentelemetry_http_client_curl
-                          PUBLIC opentelemetry_ext CURL::libcurl)
+    target_link_libraries(
+      opentelemetry_http_client_curl
+      PUBLIC opentelemetry_ext
+      PRIVATE CURL::libcurl)
   else()
     target_include_directories(opentelemetry_http_client_curl
                                INTERFACE "${CURL_INCLUDE_DIRS}")
-    target_link_libraries(opentelemetry_http_client_curl
-                          PUBLIC opentelemetry_ext ${CURL_LIBRARIES})
+    target_link_libraries(
+      opentelemetry_http_client_curl
+      PUBLIC opentelemetry_ext
+      PRIVATE ${CURL_LIBRARIES})
   endif()
 
   install(

